02.Capstone_Project_Details/SquadScript_AyushKumarTripathi.md

```diff
@@ -0,0 +1,50 @@
+# Project Initial Report
+
+## Ayush Kumar Tripathi 12103634
+
+### Project Name: SquadScript
+
+- **Which problem Project is Solving?**
+  - SquadScript is an innovative coding platform designed to foster collaborative learning and development. It allows up to four people to come together from anywhere in the world and code together while engaging in real-time conversation.
+
+- **What are existing similar solutions?**
+  - 1. **Visual Studio Code Live Share:**
+    - *Similar Features:* Collaborative coding, chat, live editing.
+  - 2. **CodePen:**
+    - *Similar Features:* Online code editor, live output.
+  - 3. **Zoom with Code Editor:**
+    - *Similar Features:* Video calls with coding collaboration.
+
+- **How do you want to solve it?**
+  - SquadScript will provide an integrated platform with a focus on collaborative coding. It aims to enhance the coding experience by offering features like real-time code editing, video calls, chat, and additional tools such as a whiteboard for better collaboration.
+
+- **List of features you want to implement:**
+  - 1. **Collaborative Coding:**
+    - Video calls for up to 4 participants.
+    - Chat functionality for communication.
+    - Dual code editor - personal and main.
+  - 2. **Integration with GitHub:**
+    - Seamless integration with GitHub for version control repositories management.
+  - 3. **Live Output Panel:**
+    - Real-time output for HTML, CSS, JS code.
+  - 4. **User Management:**
+    - Login and signup functionality.
+    - User profile pages.
+  - 5. **Code File Handling:**
+    - Uploading code files to the code editor.
+    - Multiple people editing the code simultaneously.
+  - 6. **Editor Tabs:**
+    - Opening multiple editor tabs together.
+  - 7. **Recording and Saving:**
+    - Recording collaborative coding sessions.
+    - Saving recordings locally.
+
+## Steps to follow when you raise a PR.
```

Comment:
No need to add,

## Steps to follow when you raise a PR.

- Fork the squadscript_repo [https://github.com/AyushKumarTripathi/squadscript_repo]
- Clone it from your GitHub account to your local
- Add a readme file with `AyushKumarTripathi_SquadScript.md` format
- Add descriptive responses for the above questions.
- Push your changes to GitHub.
- Create a pull request from your account to the squadscript_repo [https://github.com/AyushKumarTripathi/squadscript_repo]
- After the review, make changes or merge the pull request.


these were for you all to follow, you can remove these many lines from here.